whylog/assistant/regex_assistant/__init__.py

```diff
@@ -0,0 +1,64 @@
+from whylog.assistant import AbstractAssistant
+from whylog.assistant.regex_assistant.regex_object import RegexObject
+from whylog.assistant.spans_finding import find_date_spans
+
+
+class RegexAssistant(AbstractAssistant):
+    """
+    Responsible for creating regex for each line of Rule.
+    RegexAssistant helps user to write regex, it can also propose regex.
+    One RegexAssistant per one entering Rule.
+    """
+
+    def __init__(self):
+        self.regex_objects = {}
+
+    def add_line(self, line_id, line_object):
```

Comment:
In RegexAssistent line is represented by regex_object, that keeps information about line, regex and "bricks" (Spans) needed to create and update regex. In Teacher, line keeps information about line, constraints and regex, but not bricks needed for build regex.